Compilation fails with Visual Studio 2015

The current master branch of ArrayFire does not compile with Visual Studio 2015:
```
src\backend\common/half.hpp(701): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(702): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(705): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(706): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(707): error : a constexpr function must contain exactly one return statement
src\backend\common/half.hpp(708): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(709): error : a constexpr function must contain exactly one return statement
src\backend\common/half.hpp(710): error : a constexpr function must contain exactly one return statement
src\backend\common/half.hpp(712): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(713): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(714): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(715): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(716): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(717): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(718): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(719): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(720): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(721): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(723): error : statement may not appear in a constexpr function
src\backend\common/half.hpp(724): error : a constexpr function must contain exactly one return statement

src\backend\common/unique_handle.hpp(77): error : statement may not appear in a constexpr constructor
            detected during:
              instantiation of "common::unique_handle<T>::unique_handle(common::unique_handle<T> &&) [with T=cusparseMatDescr_t]"
  (111): here
              instantiation of "common::unique_handle<T> common::make_handle<T>() [with T=cusparseMatDescr_t]"
  src/backend/cuda/sparse_arith.cu(146): here
              instantiation of "common::SparseArray<T> cuda::arithOp<T,op>(const common::SparseArray<T> &, const common::SparseArray<T> &) [with T=float, op=af_add_t]"
  src/backend/cuda/sparse_arith.cu(224): here

```

Solved by removing ``constexpr`` from ``half2int`` and from the move constructor of ``unique_handle``.

Also a few worrisome warning:
```
src\backend\common/half.hpp(261): warning C4146: unary minus operator applied to unsigned type, result still unsigned
src\backend\common/half.hpp(262): warning C4146: unary minus operator applied to unsigned type, result still unsigned
src\backend\common/half.hpp(709): warning C4146: unary minus operator applied to unsigned type, result still unsigned
src\backend\common/half.hpp(724): warning C4146: unary minus operator applied to unsigned type, result still unsigned
src\backend\common/unique_handle.hpp(102): warning : constexpr non-static member function will not be implicitly 'const' in C++14
```

There were quite a few other warnings of potentially lossy conversions from ``dim_t`` to ``int`` and ``double`` to ``float``. I assume these are mostly harmless, but here you go if you are interested:
```
src/api/c/error.cpp(38): warning C4800: 'int': forcing value to bool 'true' or 'false' (performance warning)
src/backend/common/DependencyModule.cpp(90): warning C4800: 'const LibHandle': forcing value to bool 'true' or 'false' (performance warning)
src/api/c/print.cpp(102): warning C4267: 'argument': conversion from 'size_t' to 'unsigned int', possible loss of data
src/api/c/stream.cpp(53): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
src/backend/common/ArrayInfo.cpp(124): warning C4267: 'argument': conversion from 'size_t' to 'const unsigned int', possible loss of data
src/backend/common/ArrayInfo.cpp(125): warning C4267: 'argument': conversion from 'size_t' to 'const unsigned int', possible loss of data
src/backend/common/jit/Node.cpp(30): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/copy.cpp(68): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/copy.cpp(74): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/copy.cpp(81): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/copy.cpp(94): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/device_manager.cpp(536): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/device_manager.cpp(602): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/kernel/bilateral.hpp(58): warning C4267: 'argument': conversion from 'size_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/flood_fill.hpp(60): warning C4267: 'argument': conversion from 'size_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/mean.hpp(472): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/kernel/mean.hpp(561): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/kernel/medfilt.hpp(66): warning C4267: 'argument': conversion from 'size_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/memcopy.hpp(73): warning C4267: 'initializing': conversion from 'size_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/memcopy.hpp(74): warning C4267: 'initializing': conversion from 'size_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/nearest_neighbour.hpp(169): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/orb.hpp(288): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(835): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(837): warning C4267: 'initializing': conversion from 'size_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(851): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(853): warning C4267: 'initializing': conversion from 'size_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(854): warning C4267: 'argument': conversion from 'size_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(865): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(872): warning C4267: 'argument': conversion from 'size_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(876): warning C4267: 'argument': conversion from 'size_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(891): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(898): warning C4267: 'argument': conversion from 'size_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(902): warning C4267: 'argument': conversion from 'size_t' to 'cuda::uint', possible loss of data
src/backend/cuda/nvrtc/cache.cpp(271): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
src/api/c/det.cpp(33): warning C4244: 'initializing': conversion from 'const dim_t' to 'const int', possible loss of data
src/api/c/det.cpp(33): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/api/c/hist.cpp(44): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/api/c/imageio2.cpp(345): warning C4244: 'initializing': conversion from 'const dim_t' to 'const cuda::uint', possible loss of data
src/api/c/imageio2.cpp(345): warning C4244: 'initializing': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/api/c/imageio2.cpp(346): warning C4244: 'initializing': conversion from 'const dim_t' to 'const cuda::uint', possible loss of data
src/api/c/imageio2.cpp(346): warning C4244: 'initializing': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/api/c/imageio2.cpp(423): warning C4244: 'initializing': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/api/c/imageio2.cpp(424): warning C4244: 'initializing': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/api/c/imageio2.cpp(88): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/api/c/imageio_helper.h(88): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/api/c/imageio_helper.h(89): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/api/c/imageio_helper.h(90): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/api/c/imageio_helper.h(91): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/api/c/imageio_helper.h(93): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/api/c/imageio_helper.h(94): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/api/c/imageio_helper.h(95): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/api/c/imageio_helper.h(97): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/api/c/pinverse.cpp(125): warning C4244: 'argument': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/api/c/pinverse.cpp(129): warning C4244: 'argument': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/api/c/pinverse.cpp(72): warning C4244: 'argument': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/api/c/pinverse.cpp(74): warning C4244: 'argument': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/api/c/pinverse.cpp(76): warning C4244: 'argument': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/api/c/pinverse.cpp(78): warning C4244: 'argument': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/api/c/window.cpp(111): warning C4244: 'argument': conversion from 'double' to 'const float', possible loss of data
src/api/c/window.cpp(151): warning C4244: 'argument': conversion from 'double' to 'const float', possible loss of data
src/api/c/window.cpp(196): warning C4244: 'argument': conversion from 'double' to 'const float', possible loss of data
src/api/cpp/lapack.cpp(135): warning C4244: 'return': conversion from 'double' to 'float', possible loss of data
src/api/cpp/seq.cpp(27): warning C4244: '=': conversion from 'double' to 'std::size_t', possible loss of data
src/api/cpp/timing.cpp(101): warning C4244: 'argument': conversion from 'double' to 'float', possible loss of data
src/backend/common/ArrayInfo.cpp(164): warning C4244: '=': conversion from 'const double' to 'dim_t', possible loss of data
src/backend/common/ArrayInfo.cpp(166): warning C4244: '=': conversion from 'double' to 'dim_t', possible loss of data
src/backend/common/ArrayInfo.cpp(181): warning C4244: '*=': conversion from 'const double' to 'dim_t', possible loss of data
src/backend/common/DefaultMemoryManager.cpp(123): warning C4244: '=': conversion from 'double' to 'std::size_t', possible loss of data
src/backend/common/dim4.cpp(113): warning C4244: '=': conversion from 'double' to 'std::size_t', possible loss of data
src/backend/common/dim4.cpp(115): warning C4244: '=': conversion from 'double' to 'std::size_t', possible loss of data
src/backend/cuda/blas.cu(220): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(220): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(237): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(287): warning C4244: 'initializing': conversion from 'const dim_t' to 'const int', possible loss of data
src/backend/cuda/blas.cu(287): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(288): warning C4244: 'initializing': conversion from 'const dim_t' to 'const int', possible loss of data
src/backend/cuda/blas.cu(288): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(289): warning C4244: 'initializing': conversion from 'const dim_t' to 'const int', possible loss of data
src/backend/cuda/blas.cu(289): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(301): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(317): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(318): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(319): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(321): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(345): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(366): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(367): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/blas.cu(374): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/diagonal.cpp(23): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/diagonal.cpp(24): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/fast_pyramid.cpp(31): warning C4244: 'initializing': conversion from 'const dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/fast_pyramid.cpp(36): warning C4244: '/=': conversion from 'const float' to 'unsigned int', possible loss of data
src/backend/cuda/fast_pyramid.cpp(53): warning C4244: '=': conversion from 'float' to 'unsigned int', possible loss of data
src/backend/cuda/fast_pyramid.cpp(74): warning C4244: 'argument': conversion from 'float' to 'dim_t', possible loss of data
src/backend/cuda/fast_pyramid.cpp(90): warning C4244: 'initializing': conversion from 'float' to 'unsigned int', possible loss of data
src/backend/cuda/fft.cu(108): warning C4244: '*=': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/fft.cu(113): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/fft.cu(136): warning C4244: '*=': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/fft.cu(143): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/fft.cu(63): warning C4244: '=': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/fft.cu(78): warning C4244: '*=': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/fft.cu(80): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/homography.cu(34): warning C4244: 'initializing': conversion from 'const dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/homography.cu(34): warning C4244: 'initializing': conversion from 'const dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/jit.cpp(288): warning C4244: '=': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/jit.cpp(292): warning C4244: 'initializing': conversion from '__int64' to 'int', possible loss of data
src/backend/cuda/jit.cpp(298): warning C4244: '=': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/jit.cpp(299): warning C4244: '=': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/jit.cpp(301): warning C4244: '=': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/jit.cpp(302): warning C4244: '=': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/jit.cpp(304): warning C4244: '=': conversion from '__int64' to 'int', possible loss of data
src/backend/cuda/jit.cpp(308): warning C4244: 'initializing': conversion from '__int64' to 'int', possible loss of data
src/backend/cuda/kernel/anisotropic_diffusion.hpp(41): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/anisotropic_diffusion.hpp(42): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/anisotropic_diffusion.hpp(44): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/approx.hpp(39): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/approx.hpp(40): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/approx.hpp(69): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/approx.hpp(70): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/approx.hpp(71): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/bilateral.hpp(36): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/bilateral.hpp(37): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/bilateral.hpp(39): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/canny.hpp(41): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/canny.hpp(42): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/canny.hpp(45): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/canny.hpp(72): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/canny.hpp(73): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/canny.hpp(76): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/convolve.hpp(302): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/convolve.hpp(303): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/convolve.hpp(325): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/convolve.hpp(326): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/convolve.hpp(328): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/diagonal.hpp(31): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/diagonal.hpp(32): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/diagonal.hpp(33): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/diagonal.hpp(58): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/diagonal.hpp(59): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/diagonal.hpp(60): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/fast.hpp(327): warning C4244: 'initializing': conversion from 'float' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/fast.hpp(327): warning C4244: 'initializing': conversion from 'float' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/fast.hpp(330): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/fast.hpp(395): warning C4244: '=': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/fast.hpp(396): warning C4244: '=': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/fast.hpp(405): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/fast.hpp(409): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/fast.hpp(431): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/flood_fill.hpp(67): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/gradient.hpp(35): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/gradient.hpp(36): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/gradient.hpp(37): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/harris.hpp(176): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/harris.hpp(216): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/harris.hpp(217): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/harris.hpp(237): warning C4244: 'initializing': conversion from 'float' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/harris.hpp(251): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/harris.hpp(253): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/harris.hpp(260): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/harris.hpp(292): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/harris.hpp(39): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data
src/backend/cuda/kernel/identity.hpp(31): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/identity.hpp(32): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/identity.hpp(33): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/iota.hpp(37): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/iota.hpp(38): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/iota.hpp(40): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(104): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(110): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(132): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(133): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(173): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(198): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(199): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(209): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(241): warning C4244: '+=': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(38): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(61): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/ireduce.hpp(78): warning C4244: '*=': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/lookup.hpp(45): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/lookup.hpp(60): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/lookup.hpp(61): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/lookup.hpp(63): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/mean.hpp(174): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/mean.hpp(178): warning C4244: 'argument': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/mean.hpp(183): warning C4244: 'argument': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/mean.hpp(188): warning C4244: 'argument': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/mean.hpp(193): warning C4244: 'argument': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/mean.hpp(204): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/mean.hpp(359): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/mean.hpp(361): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/mean.hpp(391): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/mean.hpp(392): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/mean.hpp(434): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/mean.hpp(464): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/mean.hpp(465): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/mean.hpp(529): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/mean.hpp(549): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/mean.hpp(550): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/medfilt.hpp(40): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/medfilt.hpp(41): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/medfilt.hpp(43): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/medfilt.hpp(60): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/medfilt.hpp(62): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/memcopy.hpp(44): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/memcopy.hpp(45): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/memcopy.hpp(47): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/memcopy.hpp(76): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/memcopy.hpp(83): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/memcopy.hpp(84): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/memcopy.hpp(85): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/memcopy.hpp(86): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/morph.hpp(33): warning C4244: 'initializing': conversion from 'dim_t' to 'const int', possible loss of data
src/backend/cuda/kernel/morph.hpp(33): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/morph.hpp(48): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/morph.hpp(49): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/morph.hpp(51): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/morph.hpp(68): warning C4244: 'initializing': conversion from 'dim_t' to 'const int', possible loss of data
src/backend/cuda/kernel/morph.hpp(68): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/morph.hpp(82): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/morph.hpp(83): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/morph.hpp(84): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/morph.hpp(85): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/nearest_neighbour.hpp(156): warning C4244: 'initializing': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/nearest_neighbour.hpp(156): warning C4244: 'initializing': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/nearest_neighbour.hpp(176): warning C4244: 'argument': conversion from 'const dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/nearest_neighbour.hpp(176): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/nearest_neighbour.hpp(180): warning C4244: 'argument': conversion from 'const dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/nearest_neighbour.hpp(180): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/kernel/orb.hpp(40): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data
src/backend/cuda/kernel/pad_array_borders.hpp(37): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/pad_array_borders.hpp(38): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/pad_array_borders.hpp(40): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(867): warning C4244: 'initializing': conversion from 'const cuda::uintl' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(869): warning C4244: 'initializing': conversion from 'cuda::uintl' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(893): warning C4244: 'initializing': conversion from 'const cuda::uintl' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/random_engine.hpp(895): warning C4244: 'initializing': conversion from 'cuda::uintl' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/range.hpp(36): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/range.hpp(37): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/range.hpp(38): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/reduce.hpp(116): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/reduce.hpp(125): warning C4244: 'argument': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/reduce.hpp(130): warning C4244: 'argument': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/reduce.hpp(135): warning C4244: 'argument': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/reduce.hpp(140): warning C4244: 'argument': conversion from 'const dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/reduce.hpp(151): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/reduce.hpp(165): warning C4244: '*=': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/reduce.hpp(265): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/reduce.hpp(267): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/reduce.hpp(307): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/reduce.hpp(308): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/reduce.hpp(349): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/reduce.hpp(370): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/reduce.hpp(371): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/reduce.hpp(381): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/regions.hpp(353): warning C4244: 'initializing': conversion from 'dim_t' to 'const int', possible loss of data
src/backend/cuda/kernel/regions.hpp(353): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/regions.hpp(354): warning C4244: 'initializing': conversion from 'dim_t' to 'const int', possible loss of data
src/backend/cuda/kernel/regions.hpp(354): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/regions.hpp(385): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/regions.hpp(399): warning C4244: 'initializing': conversion from 'double' to 'const int', possible loss of data
src/backend/cuda/kernel/regions.hpp(399): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data
src/backend/cuda/kernel/regions.hpp(399): warning C4244: 'initializing': conversion from 'float' to 'const int', possible loss of data
src/backend/cuda/kernel/regions.hpp(399): warning C4244: 'initializing': conversion from 'float' to 'int', possible loss of data
src/backend/cuda/kernel/scan_dim_by_key_impl.hpp(103): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/scan_dim_by_key_impl.hpp(131): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/scan_dim_by_key_impl.hpp(143): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/scan_dim_by_key_impl.hpp(48): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/scan_dim_by_key_impl.hpp(50): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/scan_dim_by_key_impl.hpp(72): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/scan_dim_by_key_impl.hpp(74): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/scan_dim_by_key_impl.hpp(90): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/scan_dim_by_key_impl.hpp(92): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/scan_first_by_key_impl.hpp(121): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/scan_first_by_key_impl.hpp(45): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/scan_first_by_key_impl.hpp(47): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/scan_first_by_key_impl.hpp(66): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/scan_first_by_key_impl.hpp(68): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/scan_first_by_key_impl.hpp(84): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/scan_first_by_key_impl.hpp(85): warning C4244: 'initializing': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/sort.hpp(27): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort.hpp(29): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort.hpp(31): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort.hpp(66): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort.hpp(69): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort.hpp(74): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(26): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(27): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(29): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(30): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(32): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(33): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(35): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(68): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(69): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(76): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(77): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/sort_by_key.hpp(86): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/topk.hpp(117): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/topk.hpp(128): warning C4244: 'argument': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/topk.hpp(132): warning C4244: 'argument': conversion from 'dim_t' to 'cuda::uint', possible loss of data
src/backend/cuda/kernel/topk.hpp(147): warning C4244: 'argument': conversion from 'float' to 'dim_t', possible loss of data
src/backend/cuda/kernel/topk.hpp(148): warning C4244: 'argument': conversion from 'float' to 'dim_t', possible loss of data
src/backend/cuda/kernel/topk.hpp(93): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/topk.hpp(94): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/transpose_inplace.hpp(43): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/transpose_inplace.hpp(44): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/transpose_inplace.hpp(45): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/kernel/triangle.hpp(38): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/triangle.hpp(39): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/kernel/triangle.hpp(40): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/lookup.cpp(27): warning C4244: 'argument': conversion from 'dim_t' to 'const unsigned int', possible loss of data
src/backend/cuda/lookup.cpp(34): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/math.hpp(298): warning C4244: 'argument': conversion from 'double' to 'float', possible loss of data
src/backend/cuda/math.hpp(388): warning C4244: '=': conversion from 'double' to 'float', possible loss of data
src/backend/cuda/math.hpp(389): warning C4244: '=': conversion from 'double' to 'float', possible loss of data
src/backend/cuda/medfilt.cpp(29): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/medfilt.cpp(42): warning C4244: 'argument': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/reduce_impl.hpp(144): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/reduce_impl.hpp(208): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/reduce_impl.hpp(228): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/reduce_impl.hpp(231): warning C4244: 'argument': conversion from 'dim_t' to 'const int', possible loss of data
src/backend/cuda/reduce_impl.hpp(239): warning C4244: 'argument': conversion from 'dim_t' to 'const int', possible loss of data
src/backend/cuda/reduce_impl.hpp(251): warning C4244: 'argument': conversion from 'dim_t' to 'const int', possible loss of data
src/backend/cuda/reduce_impl.hpp(285): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/reduce_impl.hpp(300): warning C4244: 'argument': conversion from 'dim_t' to 'const int', possible loss of data
src/backend/cuda/reduce_impl.hpp(66): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/reduce_impl.hpp(86): warning C4244: 'argument': conversion from 'dim_t' to 'unsigned int', possible loss of data
src/backend/cuda/reduce_impl.hpp(89): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(201): warning C4244: 'argument': conversion from 'const dim_t' to 'const int', possible loss of data
src/backend/cuda/sparse.cu(218): warning C4244: 'initializing': conversion from 'const dim_t' to 'const int', possible loss of data
src/backend/cuda/sparse.cu(218): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(219): warning C4244: 'initializing': conversion from 'const dim_t' to 'const int', possible loss of data
src/backend/cuda/sparse.cu(219): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(240): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(256): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(260): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(294): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(295): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(297): warning C4244: 'initializing': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(321): warning C4244: 'initializing': conversion from 'dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(332): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(338): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(349): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(371): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(382): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/sparse.cu(403): warning C4244: 'argument': conversion from 'const dim_t' to 'int', possible loss of data
src/backend/cuda/utility.hpp(24): warning C4244: '=': conversion from 'dim_t' to 'int', possible loss of data
```

--- a/src/backend/common/half.hpp
+++ b/src/backend/common/half.hpp
@@ -695,7 +695,7 @@ CONSTEXPR_DH inline float half2float(native_half_t value) noexcept {
 ///           value
 /// \param value The value to convert to integer
 template<std::float_round_style R, bool E, typename T>
-constexpr T half2int(native_half_t value) {
+T half2int(native_half_t value) {
     static_assert(std::is_integral<T>::value,
                   "half to int conversion only supports builtin integer types");
     unsigned int e = value & 0x7FFF;

--- a/src/backend/common/unique_handle.hpp
+++ b/src/backend/common/unique_handle.hpp
@@ -72,7 +72,7 @@ class unique_handle {
     constexpr operator const T &() const noexcept { return handle_; }
 
     unique_handle(const unique_handle &other) noexcept = delete;
-    constexpr unique_handle(unique_handle &&other) noexcept
+    unique_handle(unique_handle &&other) noexcept
         : handle_(other.handle_) {
         other.handle_ = 0;
     }

--- a/src/backend/common/unique_handle.hpp
+++ b/src/backend/common/unique_handle.hpp
@@ -72,8 +72,7 @@ class unique_handle {
     constexpr operator const T &() const noexcept { return handle_; }
 
     unique_handle(const unique_handle &other) noexcept = delete;
-    unique_handle(unique_handle &&other) noexcept
-        : handle_(other.handle_) {
+    unique_handle(unique_handle &&other) noexcept : handle_(other.handle_) {
         other.handle_ = 0;
     }
 

